packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -584,20 +583,13 @@ export interface RawResolversConfig extends RawConfig {
   internalResolversPrefix?: string;
   /**
    * @type object
-   * @default { __resolveReference: false }
+   * @default {}
    * @description If relevant internal resolvers are set to `true`, the resolver type will only be generated if the right conditions are met.
    * Enabling this allows a more correct type generation for the resolvers.
    * For example:
    * - `__isTypeOf` is generated for implementing types and union members
-   * - `__resolveReference` is generated for federation types that have at least one resolvable `@key` directive
    */
   generateInternalResolversIfNeeded?: GenerateInternalResolversIfNeededConfig;
-  /**
-   * @type boolean
-   * @default false
-   * @description Turning this flag to `true` will generate resolver signature that has only `resolveType` for interfaces, forcing developers to write inherited type resolvers in the type itself.
-   */
-  onlyResolveTypeForInterfaces?: boolean;
```

Comment:
Related https://github.com/dotansimha/graphql-code-generator/issues/5648

`onlyResolveTypeForInterfaces` is only applicable for some rare use cases mentioned [here](https://github.com/dotansimha/graphql-code-generator/issues/5648#issuecomment-790151449). This option's implementation blocks `__resolveReference` being created for Interfaces when the option is turned on.

I think it's simpler just to deprecate this option as we are moving to a new major version. If anyone complains, we can put it back as a minor version